### PR TITLE
[xeus] update to 0.24.3

### PR DESCRIPTION
**Pull request by SeekingMeaning**
**Describe the pull request**

- What does your PR fix? Fixes #

- Which triplets are supported/not supported? Have you updated the CI baseline?

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

File: ports/xeus/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: xeus
-Version: 0.24.1
+Version: 0.24.3
 Port-Version: 1
```
Comment (NancyLi1013) [inline ports/xeus/CONTROL:3]:
```suggestion
```
This field is not needed, since this is a new version update.

File: ports/xeus/portfile.cmake [hunk 2]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO QuantStack/xeus
-    REF 424b7cd177886906a59eee535b7de59088461910 # 0.24.1
-    SHA512 877ca45bf649b567a9921d3e8f0adb0299dbe956978bd6e217d0c06617cf3466d08d90d607fd33e129089472e1a96ecec78b1fc21346bc13ba268168a5a6b068
+    REF 0f6327a2782181e7ded9729abb32b7d8eb690aea # 0.24.3
+    SHA512 2c0ccd1bebf487a9a73e73ecfb74b7605756652b2a84c71e739d7b2d8923960594c025e36d75cec850c5f0e38614a20299feccea6cfbe9ea0f66bdf315af02b4
     HEAD_REF master
     PATCHES
 		Fix-Compile-nlohmann-json.patch
```
Comment (NancyLi1013) [inline ports/xeus/portfile.cmake:8]:
```suggestion
    PATCHES
        Fix-Compile-nlohmann-json.patch
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

File: ports/xeus/portfile.cmake [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-    SHA512 877ca45bf649b567a9921d3e8f0adb0299dbe956978bd6e217d0c06617cf3466d08d90d607fd33e129089472e1a96ecec78b1fc21346bc13ba268168a5a6b068
+    REF 0f6327a2782181e7ded9729abb32b7d8eb690aea # 0.24.3
+    SHA512 2c0ccd1bebf487a9a73e73ecfb74b7605756652b2a84c71e739d7b2d8923960594c025e36d75cec850c5f0e38614a20299feccea6cfbe9ea0f66bdf315af02b4
     HEAD_REF master
     PATCHES
-		Fix-Compile-nlohmann-json.patch
+        Fix-Compile-nlohmann-json.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC_LIBS)
```
Comment (NancyLi1013) [inline ports/xeus/portfile.cmake:11]:
```suggestion
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" BUILD_SHARED_LIBS)
```
Dynamic is still required to be added here.

Comment (NancyLi1013) [reply ports/xeus/portfile.cmake:11]: The OPTIONS should be like this:

```
-DXEUS_BUILD_STATIC_LIBS=${BUILD_STATIC_LIBS}
-DXEUS_BUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @SeekingMeaning.

**Comment by strega-nil** [comment]
Great, thanks @SeekingMeaning :)

**Pull request merged**